support/routes/v2/isobot/ops/botService.ts

```diff
@@ -19,13 +27,84 @@ export interface SlackPayload {
   text: string
 }
 
+// Slack only gives us 3 seconds to respond, but usually DNS resolution is fast,
+// so we will give it 2 seconds before we give up
+const DNS_TIMEOUT = 2000
+
 class BotService {
   whitelistService: WhitelistService
 
   constructor(whitelistService: WhitelistService) {
     this.whitelistService = whitelistService
   }
 
+  private checkCname(domain: string) {
+    return ResultAsync.fromPromise(
+      Promise.race([
+        dns.resolveCname(domain),
+        new Promise<null>((_, reject) =>
+          setTimeout(() => reject(null), DNS_TIMEOUT)
```

Comment:
rather than null can just put something like `timeout exceeded`

Reply:
arh unless this was a typing thing

Reply:
Ah yeah the `logger.info` would have captured this.